src/app/listener.ts

```diff
@@ -69,6 +80,17 @@ export class Listener {
       notificationChannel.send({ embeds: [embed] });
     });
 
+    // Used to handle slash commands.
+    this.container.clientService.on('interactionCreate', (interaction: Interaction) => {
+      // If it's not a command, we don't care.
+      if (!interaction.isCommand()) {
+        return;
+      }
+
+      // We only need the slash command handler.
+      slashCommands[interaction.commandName].execute({ interaction, container: this.container });
```

Comment:
The types for accessing a prop on a `Record<_,T>` using an arbitrary key are misleading, it should give `T | undefined` but `T` is given instead. Beyond the types, there also may be situations where the deployed commands on discord are de-synced with our local commands defined in the source which may trigger an undefined access error.

```suggestion
      slashCommands[interaction.commandName]?.execute({ interaction, container: this.container });
```

Reply:
I redid the Record type differently to fix this

Reply:
While the updated type makes the types correct in this scenario it makes the incorrect in others (like when iterating over entries, the value can never be nullish). I think using a `Map<K,V>` instead of an object would avoid all these issues.

Reply:
Ok, I changed this